game/ui/base/researchselect.cpp

```diff
@@ -80,31 +79,32 @@ void ResearchSelect::begin()
 			if (topic->isComplete())
 			{
 				LogInfo("Topic already complete");
-				auto message_box = mksp<MessageBox>(tr("PROJECT COMPLETE"),
-				                                    tr("This project is already complete."),
-				                                    MessageBox::ButtonOptions::Ok);
-				fw().stageQueueCommand({StageCmd::Command::PUSH, message_box});
+				fw().stageQueueCommand({StageCmd::Command::PUSH,
```

Comment:
To my eye these blocks look functionally identical?

Am I missing a change?

If just a cleanup I don't know if it's necessary, or at least should be a separate commit to functional changes.

Reply:
I just remove AlwaysEmitSelectionEvents = true and container variables for dialogues.